src/linear_algebra/multilinear.lean

```diff
@@ -437,15 +461,83 @@ instance : has_scalar R (multilinear_map R M₁ M₂) := ⟨λ c f,
 
 @[simp] lemma smul_apply (c : R) (m : Πi, M₁ i) : (c • f) m = c • f m := rfl
 
+section
+
+variables (R ι) (A : Type*) [comm_semiring A] [algebra R A] [fintype ι]
+
+/-- Given an `R`-algebra `A`, `mk_pi_algebra` is the multilinear map on `A^ι` associating
+to `m` the product of all the `m i`.
+
+TODO: This version assumes that `A` is commutative. In the most interesting case `ι = fin n`,
+it is possible to drop this assumption. -/
```

Comment:
#3770 likely needs this case. Perhaps the lemma should be called `mk_pi_comm_algebra` to reserve `mk_pi_algebra` for the non-commutative one.

 Is it sufficient to require `linear_order ι` instead of `fin n` for the non-commutative cases?

Reply:
Oh, actually you already have done this todo, so maybe just remove the comment and link to `mk_pi_algebra_fin`

Reply:
I should've marked this as WIP instead of awaiting-author. I'll fix this tonight.